cmd/injective-price-oracle/oracle.go

```diff
@@ -228,3 +294,34 @@ func oracleCmd(cmd *cli.Cmd) {
 		closer.Hold()
 	}
 }
+
+func ConnectWebSocket(ctx context.Context, websocketUrl, urlHeader string) (conn *websocket.Conn, err error) {
+	u, err := url.Parse(websocketUrl)
+	if err != nil {
+		log.Fatal("Error parsing URL:", err)
+		return &websocket.Conn{}, err
```

Comment:
this fatal will panic, it should return the error instead
```suggestion
		return &websocket.Conn{}, errors.Wrapf(err, "can not parse WS url %s: %v", websocketUrl, err)
```